src/main/webapp/availability.js

```diff
@@ -12,86 +12,103 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function getAvailability() {
+/** Creates a calendar with the Charts API and renders it on the page  */
+function createCalendar() {
     var queryString = new Array();
     window.onload = readComponents(queryString, window);
     const tutorID = queryString["tutorID"];
 
-    fetch('/availability?tutorID=' + tutorID, {method: 'GET'}).then(response => response.json()).then((timeslots) => {
+    fetch('/availability?tutorID=' + tutorID, {method: 'GET'}).then(response => response.json()).then((timeslots) => {  
         if(timeslots.error) {
             var message = document.createElement("p");
             p.innerText = timeslots.error;
-            document.getElementById('timeslots').appendChild(message);
+            document.getElementById('calendar').appendChild(message);
             return;
         }
 
-        if (Object.keys(timeslots).length != 0) {
-            timeslots.forEach((timeslot) => {
-                document.getElementById('timeslots').appendChild(createTimeSlotBox(timeslot, tutorID));
-            });
-        } else {
-            var timeslotsContainer = document.getElementById('timeslots');
-            var errorMessage = document.createElement("p");
-            errorMessage.innerText = "This user has not set any available timeslots";
-            timeslotsContainer.appendChild(errorMessage);
+        // Don't create a calendar if there are no available timeslots
+        if (timeslots === undefined || timeslots.length == 0) {
```

Comment:
Could you keep the error message that is displayed if there are no timeslots?

Reply:
Sure. I forgot to add it earlier.